github/Requester.py

```diff
@@ -1235,18 +1235,18 @@ def __makeAbsoluteUrl(self, url: str) -> str:
     def __createConnection(
         self, hostname: str | None = None
     ) -> HTTPRequestsConnectionClass | HTTPSRequestsConnectionClass:
-        if self.__persist and self.__connection is not None and hostname is not None and hostname == self.__hostname:
+        if hostname is None:
+            hostname = self.__hostname
+
+        if self.__persist and self.__connection is not None and hostname == self.__connection.host:
             return self.__connection
 
         with self.__connection_lock:
-            if self.__connection is not None and hostname is not None and hostname == self.__hostname:
-                if self.__persist:
-                    return self.__connection
```

Comment:
I deleted this because it seems identical to the check above and I couldn't see a reason to keep it. Am I missing something?

Reply:
This condition is protected by the lock, above condition is not. This has to be carefully evaluated in the context of multi-threaded execution. I'd say we keep this as this seems to be an optimization and not a fix of a bug.